Sources/KlaviyoSwift/Klaviyo.swift

```diff
@@ -259,13 +279,175 @@ public class Klaviyo: NSObject  {
                 , "Property values must be of NSString, NSNumber, NSNull, NSDictionary, NSDate, or NSURL. Got: \(String(describing: properties[k as! NSCopying]))")
         }
     }
+}
+
+
+
+/// The main interface for the Klaviyo SDK.
+/// Create a new instance as follows:
+///
+/// ```swift
+/// let sdk = KlaviyoSDK()
+/// sdk.initialize(apiKey: "myapikey")
+/// ```
+///
+/// From there you can you can call the additional methods below to track events and profile.
+@_spi(KlaviyoPrivate)
+public struct KlaviyoSDK {
+    
+    /// Default initializer for the Klaviyo SDK.
+    @_spi(KlaviyoPrivate)
+    public init() {
+        
+    }
+    
+    private var state: KlaviyoState {
+        get {
+            environment.analytics.state()
+        }
+    }
     
-    private func dispatchOnMainThread(action: KlaviyoAction) {
-        Task {
-            await MainActor.run {
-                _ = environment.analytics.store.send(action)
-            }
+    /// Returns the email for the current user, if any.
+    @_spi(KlaviyoPrivate)
+    public var email: String? {
+        get {
```

Comment:
Nit: Can we drop the `get`?

Reply:
Yah good point!

Reply:
So in swift a var with just get defined cannot be assigned at all, right?

Reply:
So what Noah initially had was a regular computed property which had a getter and it comes with an optional setter. However, if you know you are never going to set a value to it you can use read only computed properties that only have getters and the way you do that is by dropping the 'get`.